broker/src/test/java/io/camunda/zeebe/broker/system/partitions/impl/perf/TestState.java

```diff
@@ -80,39 +80,55 @@ private ZeebeRocksDbFactory<ZbColumnFamilies> createDbFactory() {
   }
 
   private void generateDb(
-      final ZeebeTransactionDb<ZbColumnFamilies> db,
-      final int keyCount,
-      final int keySize,
-      final int valueSize) {
+      final Path path, final ZeebeTransactionDb<ZbColumnFamilies> db, final long sizeInBytes) {
     final var txn = db.createContext();
+    final var columns =
+        Arrays.stream(ZbColumnFamilies.values())
+            .map(col -> db.createColumnFamily(col, txn, new DbString(), new DbString()))
+            .toList();
+    final var random = ThreadLocalRandom.current();
 
-    Arrays.stream(ZbColumnFamilies.values())
-        .forEach(col -> generateColumn(db, keyCount, keySize, valueSize, col, txn));
+    do {
+      final var column = columns.get(random.nextInt(columns.size()));
+      insertDataForColumn(column);
+    } while (computeSnapshotSize(path) < sizeInBytes);
   }
 
-  private void generateColumn(
-      final ZeebeTransactionDb<ZbColumnFamilies> db,
-      final int keyCount,
-      final int keySize,
-      final int valueSize,
-      final ZbColumnFamilies col,
-      final TransactionContext txn) {
-    final var column = db.createColumnFamily(col, txn, new DbString(), new DbString());
-    for (int i = 0; i < keyCount; i++) {
-      column.insert(generateData(keySize), generateData(valueSize));
+  private void insertDataForColumn(final ColumnFamily<DbString, DbString> column) {
+    for (int i = 0; i < COLUMN_BATCH_INSERT; i++) {
+      column.insert(generateData(), generateData());
     }
   }
 
-  private DbString generateData(final int size) {
-    final var random = ThreadLocalRandom.current();
-    final var buffer = new byte[size];
+  private DbString generateData() {
+    final var buffer = new byte[KEY_VALUE_SIZE];
     final var data = new DbString();
-    random.nextBytes(buffer);
+    ThreadLocalRandom.current().nextBytes(buffer);
     data.wrapBuffer(new UnsafeBuffer(buffer));
 
     return data;
   }
 
+  private static long computeSnapshotSize(final Path root) {
+    try (final var files = Files.walk(root)) {
```

Comment:
The property is (according to their wiki) an estimate of the live SST data. I have no idea how accurate it is, that's why I went with checking the file size :shrug: Since it includes L0, I expect it would also fluctuate just as much if we don't stop background work somehow.

Reply:
Good point about taking a checkpoint, but here we're not taking a checkpoint, so it shouldn't have any impact.